integrations/langfuse/src/haystack_integrations/tracing/langfuse/tracer.py

```diff
@@ -262,13 +277,20 @@ def create_span(self, context: SpanContext) -> LangfuseSpan:
             return LangfuseSpan(context.parent_span.raw_span().span(name=context.name))
 
     def handle(self, span: LangfuseSpan, component_type: Optional[str]) -> None:
+        # If the span is at the pipeline level, we add input and output keys to the span
+        at_pipeline_level = span.get_data().get(_PIPELINE_INPUT_KEY) is not None
+        if at_pipeline_level:
+            span.raw_span().update(
+                input=span.get_data().get(_PIPELINE_INPUT_KEY), output=span.get_data().get(_PIPELINE_OUTPUT_KEY)
+            )
+
         if component_type in _SUPPORTED_GENERATORS:
-            meta = span._data.get(_COMPONENT_OUTPUT_KEY, {}).get("meta")
-            if meta:
-                m = meta[0]
-                span._span.update(usage=m.get("usage") or None, model=m.get("model"))
-        elif component_type in _SUPPORTED_CHAT_GENERATORS:
-            replies = span._data.get(_COMPONENT_OUTPUT_KEY, {}).get("replies")
+            meta = span.get_data().get(_COMPONENT_OUTPUT_KEY, {}).get("meta")
+            if meta is not None:
```

Comment:
tbh this is a bit of nitpick but why not use `if meta` as that guarantees non empty list response to access on line 290 as you do.

Reply:
i.e. what if meta is []

Reply:
good point, will change back